### PR TITLE
Page Table extract

### PR DESCRIPTION
**Pull request by ederign**
First part of the page table/simple table refactoring.

Moving it to a new project gwt client only.

### CODE REVIEW

**Comment by ederign** [comment]
Related PR's:
https://github.com/droolsjbpm/jbpm-console-ng/pull/371
https://github.com/droolsjbpm/kie-wb-common/pull/314
https://github.com/droolsjbpm/drools-wb/pull/146
https://github.com/uberfire/uberfire-extensions/pull/242

File: uberfire-widgets/uberfire-widgets-table/src/main/java/org/uberfire/ext/widgets/table/client/PagedTableHelper.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package org.uberfire.ext.widgets.table.client;
```
Comment (manstis) [inline uberfire-widgets/uberfire-widgets-table/src/main/java/org/uberfire/ext/widgets/table/client/PagedTableHelper.java]: Missing license.

File: uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/tables/PagedTable.java [hunk 2]
```diff
@@ -28,9 +28,13 @@
...
+import org.uberfire.ext.widgets.table.client.PagedTableHelper;
+import org.uberfire.ext.widgets.table.client.UberfireSimplePager;
 
 /**
- * Widget that shows rows of paged data.
+ * Paged Table Widget that stores user preferences.
+ * If you doesn't need persist the preferences,
+ * take a look at UberfirePagedTable.
  */
 public class PagedTable<T>
```
Comment (manstis) [inline uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/tables/PagedTable.java]: Should this be moved too?

File: uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/tables/SimpleTable.java [hunk 3]
```diff
@@ -16,53 +16,26 @@
...
+import javax.inject.Inject;
+import java.util.List;
 
 /**
  * A composite Widget that shows rows of data (not-paged) and a "column picker"
  * to allow columns to be hidden from view. Columns can also be sorted.
+ * User preferences are persisted. If you need a client only version
+ * of this widget take a look at UberfireSimpleTable.
  */
 public class SimpleTable<T>
```
Comment (manstis) [inline uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/tables/SimpleTable.java]: Should this be moved too?

File: uberfire-widgets/uberfire-widgets-table/src/main/java/org/uberfire/ext/widgets/table/client/UberfireColumnPicker.java [hunk 4]
```diff
@@ -0,0 +1,257 @@
+package org.uberfire.ext.widgets.table.client;
```
Comment (manstis) [inline uberfire-widgets/uberfire-widgets-table/src/main/java/org/uberfire/ext/widgets/table/client/UberfireColumnPicker.java]: Missing license.

File: uberfire-widgets/uberfire-widgets-table/src/main/java/org/uberfire/ext/widgets/table/client/UberfirePagedTable.java [hunk 5]
```diff
@@ -0,0 +1,123 @@
+package org.uberfire.ext.widgets.table.client;
```
Comment (manstis) [inline uberfire-widgets/uberfire-widgets-table/src/main/java/org/uberfire/ext/widgets/table/client/UberfirePagedTable.java]: Missing license.

File: uberfire-widgets/uberfire-widgets-table/src/main/java/org/uberfire/ext/widgets/table/client/UberfireSimpleTable.java [hunk 6]
```diff
@@ -0,0 +1,401 @@
+package org.uberfire.ext.widgets.table.client;
```
Comment (manstis) [inline uberfire-widgets/uberfire-widgets-table/src/main/java/org/uberfire/ext/widgets/table/client/UberfireSimpleTable.java]: Missing license.

File: uberfire-widgets/uberfire-widgets-table/src/main/java/org/uberfire/ext/widgets/table/client/UberfireSimpleTable.java [hunk 7]
```diff
@@ -0,0 +1,401 @@
...
+    protected UberfireColumnPicker<T> columnPicker;
+
+    public UberfireSimpleTable() {
+        GWT.log( "UF SIMPLE TABLE" );
+        setupDataGrid( null );
+        setupGridTable();
+    }
+
+    public UberfireSimpleTable( final ProvidesKey<T> providesKey ) {
+        GWT.log( "UF SIMPLE TABLE" );
```
Comment (manstis) [inline uberfire-widgets/uberfire-widgets-table/src/main/java/org/uberfire/ext/widgets/table/client/UberfireSimpleTable.java]: Do we need this logging? It doesn't seem to add any value.

**Comment by manstis** [comment]
Does the webapp's ```gwt.xml`` file need an additional ```inherit```?

File: uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/tables/PagedTable.java (hunk 2 above)
Comment (ederign) [inline uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/tables/PagedTable.java]: Paged Table == Uberfire Paged Table + Preferences Persistence.

In the future with the new Preferences Persistence  (part of qmx/porcelli work) we can hook that API to UberfireTable and probably remove this class.

File: uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/tables/SimpleTable.java (hunk 3 above)
Comment (ederign) [inline uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/tables/SimpleTable.java]: same as above.

**Comment by ederign** [comment]
@manstis about "Does the webapp's gwt.xml`` file need an additionalinherit```?".

I don't think so. All of our current projects will still use PagedTable widget from commons.

File: uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/tables/PagedTable.java (hunk 2 above)
Comment (manstis) [inline uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/tables/PagedTable.java]: OK, thanks for clarifying.

File: uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/tables/SimpleTable.java (hunk 3 above)
Comment (manstis) [inline uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/tables/SimpleTable.java]: OK, thanks for clarifying.

**Comment by manstis** [comment]
@ederign OK, thanks for clarifying ```gwt.xml``` question.

+1 to merge (when missing licenses are added).

File: uberfire-widgets/uberfire-widgets-table/src/main/java/org/uberfire/ext/widgets/table/client/UberfireColumnPicker.java [hunk 8]
```diff
@@ -0,0 +1,272 @@
+/*
+ * Copyright 2016 JBoss, by Red Hat, Inc
```
Comment (ederign) [inline uberfire-widgets/uberfire-widgets-table/src/main/java/org/uberfire/ext/widgets/table/client/UberfireColumnPicker.java]: here!

File: uberfire-widgets/uberfire-widgets-table/src/main/java/org/uberfire/ext/widgets/table/client/UberfireColumnPicker.java [hunk 9]
```diff
@@ -0,0 +1,272 @@
...
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (ederign) [inline uberfire-widgets/uberfire-widgets-table/src/main/java/org/uberfire/ext/widgets/table/client/UberfireColumnPicker.java]: here!

File: uberfire-widgets/uberfire-widgets-table/src/main/java/org/uberfire/ext/widgets/table/client/UberfirePagedTable.java [hunk 10]
```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2016 JBoss, by Red Hat, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (ederign) [inline uberfire-widgets/uberfire-widgets-table/src/main/java/org/uberfire/ext/widgets/table/client/UberfirePagedTable.java]: here!

File: uberfire-widgets/uberfire-widgets-table/src/main/java/org/uberfire/ext/widgets/table/client/PagedTableHelper.java [hunk 11]
```diff
@@ -0,0 +1,49 @@
+/*
```
Comment (ederign) [inline uberfire-widgets/uberfire-widgets-table/src/main/java/org/uberfire/ext/widgets/table/client/PagedTableHelper.java]: here!

**Comment by ederign** [comment]
@manstis licenses added. thanks for your review!

**Comment by manstis** [comment]
Thanks @ederign +1 to merge.

**Comment by csadilek** [comment]
master: ad7441d65fc1a34e50b88832db692b6b3cf65136

**Pull request closed without merging**